app/src/main/java/ru/practicum/android/diploma/search/ui/SearchFragment.kt

```diff
@@ -1,18 +1,228 @@
+import ru.practicum.android.diploma.search.domain.model.Vacancy
+import ru.practicum.android.diploma.search.domain.model.VacancyList
 
 class SearchFragment : CustomFragment<FragmentSearchBinding>() {
+
+    private val viewModel by viewModel<SearchViewModel>()
+    private val vacancies = mutableListOf<Vacancy>()
+    private var onVacancyClickDebounce: ((String) -> Unit)? = null
+    private val adapter = VacancyAdapter(vacancies) { vacancy ->
+        onVacancyClickDebounce?.let { it(vacancy.id) }
+    }
+
     override fun createBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentSearchBinding {
         return FragmentSearchBinding.inflate(inflater, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        binding.vacancyList.adapter = adapter
+        binding.vacancyList.setHasFixedSize(false)
+        binding.editText.setOnFocusChangeListener { _, hasFocus ->
+            if (hasFocus && binding.editText.text.isNullOrEmpty()) {
+                render(SearchState.EmptyEditTextInFocus)
+            }
+        }
+        binding.clearButton.setOnClickListener {
+            binding.editText.setText("")
+            val inputMethodManager =
+                context?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+            inputMethodManager?.hideSoftInputFromWindow(binding.clearButton.windowToken, 0)
+            binding.editText.clearFocus()
+            render(SearchState.EmptyEditText)
+        }
+        binding.editText.doOnTextChanged { text, start, before, count ->
+            renderEditTextIconsVisibility(text)
+            viewModel.searchDebounce(text.toString())
+        }
+        viewModel.observeState().observe(viewLifecycleOwner) {
+            render(it)
+        }
+        onVacancyClickDebounce = { vacancyId ->
+            if (clickDebounce()) {
+                val direction = SearchFragmentDirections.actionSearchFragmentToVacancyFragment(vacancyId)
+                findNavController().navigate(direction)
+            }
+        }
+        binding.vacancyList.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                if (dy > 0) {
+                    val lastVisiblePosition =
+                        (binding.vacancyList.layoutManager as LinearLayoutManager).findLastVisibleItemPosition()
+                    if (lastVisiblePosition >= adapter.itemCount - 1) {
+                        viewModel.getNextPage()
+                    }
+                }
+            }
+        })
+    }
+
+    private fun render(state: SearchState, newSearch: Boolean = false) {
+        when (state) {
+            is SearchState.EmptyEditText -> {
+                setStateEmptyEditText()
+            }
```

Comment:
```suggestion
            is SearchState.EmptyEditText -> setStateEmptyEditText()
```
Если у when условие в 1 строку - скобки не нужны
https://kotlinlang.org/docs/control-flow.html#when-expression